clang/lib/CIR/Dialect/Transforms/FlattenCFG.cpp

```diff
@@ -426,11 +426,12 @@ class CIRTryOpFlattening : public mlir::OpRewritePattern<cir::TryOp> {
                      SmallVectorImpl<mlir::Block *> &landingPads) const {
     // Replace the tryOp return with a branch that jumps out of the body.
     rewriter.setInsertionPointToEnd(afterBody);
-    auto tryBodyYield = cast<cir::YieldOp>(afterBody->getTerminator());
 
     mlir::Block *beforeCatch = rewriter.getInsertionBlock();
     rewriter.setInsertionPointToEnd(beforeCatch);
-    rewriter.replaceOpWithNewOp<cir::BrOp>(tryBodyYield, afterTry);
+
+    if (auto tryBodyYield = dyn_cast<cir::YieldOp>(afterBody->getTerminator()))
```

Comment:
Add a comment here explaining that some other terminator might be present here, e.g. unreachable